build-logic/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/UpdateBranchStatus.kt

```diff
@@ -31,21 +33,19 @@ abstract class UpdateBranchStatus : DefaultTask() {
 
     @TaskAction
     fun publishBranchStatus() {
-        when (currentGitBranch()) {
+        when (project.the<ModuleIdentityExtension>().logicalBranch.get()) {
```

Comment:
Using `the<>` and accessing `project` in a task action both has a smell. The later also does not work at all with config cache. Could you instead make this a task input property and that you wire with `moduleIdentity.localBranch` when you register the task.